### PR TITLE
Feature/split screen

### PR DESCRIPTION
**Pull request by Jigar-bhaliya**

### CODE REVIEW

File: bigbluebutton-html5/imports/api/screens/client/getScreensPositions.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        newRightPosition = {
+          ...oldRightPosition,
+          component,
+          fullScreen: false,
+          visible: true,
+          url,
+        };
+      }
+
+      console.log([newLeftPosition, newRightPosition]);
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/api/screens/client/getScreensPositions.js]: remove

File: bigbluebutton-html5/imports/api/screens/client/getScreensPositions.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        ...oldRightPosition,
+        component,
+        fullScreen: false,
+        visible: true,
+        url,
+      };
+    }
+  }
+
+  console.log([newLeftPosition, newRightPosition]);
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/api/screens/client/getScreensPositions.js]: remove

File: bigbluebutton-html5/imports/ui/components/Tabs/service.js [hunk 3]
```diff
@@ -23,6 +24,10 @@ const getUsersNotAssigned = filterBreakoutUsers(currentBreakoutUsers);
 
 const takePresenterRole = () => makeCall('assignPresenter', Auth.userID);
 
+const getVideoList = () => VideoList.find({ meetingId: Auth.meetingID })
+  .fetch();
+
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/Tabs/service.js]: remove this empty line

File: bigbluebutton-html5/imports/ui/components/Tabs/service.js [hunk 4]
```diff
@@ -49,4 +54,5 @@ export default {
   getUsersNotAssigned,
   takePresenterRole,
   isSharingVideo: () => getVideoUrl(),
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/Tabs/service.js:54]: do we require this?

File: bigbluebutton-html5/imports/ui/components/external-video-player/container.jsx [hunk 5]
```diff
@@ -1,18 +1,18 @@
 import React from 'react';
 import { withTracker } from 'meteor/react-meteor-data';
 import { Session } from 'meteor/session';
-import { getVideoUrl } from './service';
+// import { getVideoUrl } from './service';
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/external-video-player/container.jsx]: remove

File: bigbluebutton-html5/imports/ui/components/join-handler/component.jsx [hunk 6]
```diff
@@ -178,19 +177,38 @@ class JoinHandler extends Component {
...
       setModOnlyMessage(response);
       logUserInfo();
 
       Tracker.autorun(async (cd) => {
-        const user = Users.findOne({ userId: Auth.userID, approved: true }, { fields: { _id: 1 } });
+        const user = Users.findOne({
+          userId: Auth.userID,
+          approved: true,
+        }, {
+          fields: { _id: 1, presenter: 1 },
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/join-handler/component.jsx]: we should remove `presenter: 1` as it was not there earlier

File: bigbluebutton-html5/imports/api/screens/client/getScreensPositions.js (hunk 1 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/api/screens/client/getScreensPositions.js]: removed

File: bigbluebutton-html5/imports/api/screens/client/getScreensPositions.js (hunk 2 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/api/screens/client/getScreensPositions.js]: removed

File: bigbluebutton-html5/imports/ui/components/Tabs/service.js (hunk 3 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/Tabs/service.js]: fixed

File: bigbluebutton-html5/imports/ui/components/Tabs/service.js (hunk 4 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/Tabs/service.js:54]: it is used in many default components, removing this line requires unnecessary editing in unused components.

File: bigbluebutton-html5/imports/ui/components/external-video-player/container.jsx (hunk 5 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/external-video-player/container.jsx]: removed

File: bigbluebutton-html5/imports/ui/components/join-handler/component.jsx (hunk 6 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/join-handler/component.jsx]: removed

File: bigbluebutton-html5/imports/api/common/client/getScreensPositions.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
+
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/api/common/client/getScreensPositions.js]: remove empty line

File: bigbluebutton-html5/imports/api/slides/server/methods/switchSlide.js [hunk 8]
```diff
@@ -16,7 +21,8 @@ export default function switchSlide(slideNumber, podId) { // TODO-- send present
   const selector = {
     meetingId,
     podId,
-    current: true,
+    id: presentationId,
+    //    current: true,
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/api/slides/server/methods/switchSlide.js]: remove

File: bigbluebutton-html5/imports/api/slides/server/methods/switchSlide.js [hunk 9]
```diff
@@ -32,6 +38,8 @@ export default function switchSlide(slideNumber, podId) { // TODO-- send present
     num: slideNumber,
   });
 
+  Logger.info(JSON.stringify(Slide));
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/api/slides/server/methods/switchSlide.js]: remove

File: bigbluebutton-html5/imports/api/slides/server/modifiers/changeCurrentSlide.js [hunk 10]
```diff
@@ -8,6 +8,7 @@ export default function changeCurrentSlide(meetingId, podId, presentationId, sli
   check(slideId, String);
   check(podId, String);
 
+  Logger.info(`meeting = ${meetingId}`);
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/api/slides/server/modifiers/changeCurrentSlide.js]: remove

File: bigbluebutton-html5/imports/ui/components/Tabs/TabsView.jsx [hunk 11]
```diff
@@ -46,7 +53,7 @@ const TabsView = ({
 
   return (
     <Fragment>
-      <aside className="primary-nav w-1/12 relative">
+      <aside className={cx('primary-nav relative shadow-2xl', { 'w-1/12': isMenuOpen, 'w-32': !isMenuOpen })}>
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/Tabs/TabsView.jsx]: make it multi-line

File: bigbluebutton-html5/imports/ui/components/Tabs/TabsView.jsx [hunk 12]
```diff
@@ -74,10 +81,16 @@ const TabsView = ({
           />
         </div>
       </aside>
-      <aside className="secondary-nav w-3/12 bg-gray-100 flex flex-col">
+      <aside className={cx('secondary-nav w-3/12 bg-gray-100 flex flex-col', { hidden: !isMenuOpen })}>
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/Tabs/TabsView.jsx]: multi-line

File: bigbluebutton-html5/imports/ui/components/Tabs/components/Slide/SlideView.jsx [hunk 13]
```diff
@@ -1,13 +1,15 @@
...
 
 const SlideView = ({
   name,
   image,
   handleSlideClick,
   pageNum,
+  selectedSlide,
 }) => (
-  <li className="p-3 bg-gray-200 hover:bg-gray-300">
+  <li className={cx('p-3 bg-gray-200 hover:bg-gray-300', { 'bg-gray-300': selectedSlide === pageNum })}>
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/Tabs/components/Slide/SlideView.jsx]: multi-line

File: bigbluebutton-html5/imports/ui/components/Tabs/components/TabPresentationContent/TabPresentationContent.jsx [hunk 14]
```diff
@@ -13,17 +13,48 @@ const TabPresentationContent = ({
...
       <div className="w-full py-3 flex flex-col" id="#Link2">
-        <TabPositionButtonGroup />
+        <TabPositionButtonGroupContainer
+          renderComponent="presentation"
+          selectedSlide={selectedSlide}
+          otherParams={{
+            presentationId,
+            slideId,
+          }}
+
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/Tabs/components/TabPresentationContent/TabPresentationContent.jsx]: remove empty line

File: bigbluebutton-html5/imports/ui/components/Tabs/components/TabVideoContent/TabVideoContent.jsx [hunk 15]
```diff
@@ -1,31 +1,70 @@
...
+      </div>
+    </div>
+    <div className="overflow-y-scroll">
+      {
+        videoList ? (
+          <ul className="list-disc">
+            {
+              videoList.map(({ _id, videoURL }) => (
+                <li
+                  className={`p-3 bg-gray-200 hover:bg-gray-300 cursor-pointer border ${_id === selectedItem._id ? 'border-red-500' : ''}`}
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/Tabs/components/TabVideoContent/TabVideoContent.jsx]: use `cx` and make it multi-line

File: bigbluebutton-html5/imports/ui/components/app/AppView.jsx [hunk 16]
```diff
@@ -19,8 +22,10 @@ const AppView = ({
         {...props}
         handleWhiteboardClick={handleWhiteboardClick}
         whiteboardOverlay={whiteboardOverlay}
+        isMenuOpen={isMenuOpen}
+        onMenuToggle={onMenuToggle}
       />
-      <section className="main-container w-11/12 py-2 px-5 flex items-center justify-between flex-col relative">
+      <section className={cx('main-container py-2 px-5 flex items-center justify-between flex-col relative', { 'w-11/12': isMenuOpen, 'w-full': !isMenuOpen })}>
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/app/AppView.jsx]: multi-line

File: bigbluebutton-html5/imports/ui/components/common/Button/Button.jsx [hunk 17]
```diff
@@ -62,11 +65,16 @@ const Button = ({
...
 
+  if (disabled) {
+    cursorClass = 'cursor-not-allowed';
+  }
+
   return (
     <button
       type="button"
-      className={cx('inline-flex items-center', buttonColor, buttonSize, buttonFontWeight)}
+      className={cx('items-center', buttonColor, buttonSize, buttonFontWeight, miscClass, cursorClass)}
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/common/Button/Button.jsx]: multi-line

File: bigbluebutton-html5/imports/ui/components/fullscreen-button/FullscreenButtonContainer.jsx [hunk 18]
```diff
@@ -4,8 +4,11 @@ import FullscreenService from './service';
...
 
-export default (props) => {
-  const handleToggleFullScreen = ref => FullscreenService.toggleFullScreen(ref);
+export default ({ onFullScreenModeChange, ...props }) => {
+  const handleToggleFullScreen = (ref) => {
+    FullscreenService.toggleFullScreen(ref);
+    onFullScreenModeChange();
+  };
   const isIphone = navigator.userAgent.match(/iPhone/i);
   return (
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/fullscreen-button/FullscreenButtonContainer.jsx:15]: empty line before return

File: bigbluebutton-html5/imports/ui/components/meeting-ended/component.jsx [hunk 19]
```diff
@@ -88,7 +88,7 @@ const propTypes = {
     formatMessage: PropTypes.func.isRequired,
   }).isRequired,
   code: PropTypes.string.isRequired,
-  reason: PropTypes.string.isRequired,
+  // reason: PropTypes.string.isRequired,
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/meeting-ended/component.jsx]: remove

File: bigbluebutton-html5/imports/ui/components/presentation/component.jsx [hunk 20]
```diff
@@ -438,12 +438,14 @@ class PresentationArea extends PureComponent {
 
   renderPresentationTopToolbar() {
     const { fitToWidth, isFullscreen } = this.state;
+    const { position } = this.props;
     return (
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/presentation/component.jsx:443]: empty line before return

File: bigbluebutton-html5/imports/ui/components/presentation/presentation-top-toolbar/PresentationTopToolbarConatiner.jsx [hunk 21]
```diff
@@ -1,9 +1,62 @@
...
+        ...rightScreen,
+        position: 'left',
+        fullScreen: true,
+        visible: true,
+      }, {
+        ...leftScreen,
+        position: 'right',
+        fullScreen: false,
+        visible: false,
+
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/presentation/presentation-top-toolbar/PresentationTopToolbarConatiner.jsx]: remove empty line

File: bigbluebutton-html5/imports/ui/components/presentation/presentation-top-toolbar/PresentationTopToolbarConatiner.jsx [hunk 22]
```diff
@@ -1,9 +1,62 @@
...
+    <PresentationTopToolbarView
+      {...props}
+      onFullScreenChange={handleFullScreenChange}
+    />
+  );
+};
+
+export default withTracker(() => {
+  const screens = Screens.find().fetch();
+  return {
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/presentation/presentation-top-toolbar/PresentationTopToolbarConatiner.jsx:56]: empty line before return

File: bigbluebutton-html5/imports/ui/components/presentation/service.js [hunk 23]
```diff
@@ -34,17 +44,16 @@ const isPresentationDownloadable = (podId) => {
   return currentPresentation.downloadable;
 };
 
-const getCurrentSlide = (podId) => {
-  const currentPresentation = getCurrentPresentation(podId);
-
+const getCurrentSlide = (podId, presentationId, selectedSlideId) => {
+  const currentPresentation = getCurrentPresentation(podId, presentationId);
   if (!currentPresentation) {
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/presentation/service.js:50]: empty line before if

File: bigbluebutton-html5/imports/ui/components/presentation/service.js [hunk 24]
```diff
@@ -181,6 +190,25 @@ const getMultiUserStatus = (whiteboardId) => {
...
+const isSplitScreen = () => {
+  if (Auth.meetingID) {
+    const isFullScreen = Screens.findOne({
+      meetingId: Auth.meetingID,
+      fullScreen: true,
+    });
+
+    return !(isFullScreen?.fullScreen);
+  }
+  return false;
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/presentation/service.js:204]: empty line before return

File: bigbluebutton-html5/imports/ui/components/presentation/service.js [hunk 25]
```diff
@@ -181,6 +190,25 @@ const getMultiUserStatus = (whiteboardId) => {
...
+};
+
+const getNewSlideId = () => {
+  if (Session.get('slideId')) {
+    return Session.get('slideId');
+  }
+  return null;
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/presentation/service.js:212]: empty line before return

File: bigbluebutton-html5/imports/ui/components/video-provider/video-button/VideoButtonView.jsx [hunk 26]
```diff
@@ -45,6 +45,7 @@ const VideoButtonView = ({
     } else {
       mountVideoPreview();
     }
+    return null;
```
Comment (piyushdhamecha) [inline bigbluebutton-html5/imports/ui/components/video-provider/video-button/VideoButtonView.jsx:49]: empty line before return

File: bigbluebutton-html5/imports/api/common/client/getScreensPositions.js (hunk 7 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/api/common/client/getScreensPositions.js]: removed

File: bigbluebutton-html5/imports/api/slides/server/methods/switchSlide.js (hunk 8 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/api/slides/server/methods/switchSlide.js]: removed

File: bigbluebutton-html5/imports/api/slides/server/methods/switchSlide.js (hunk 9 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/api/slides/server/methods/switchSlide.js]: removed

File: bigbluebutton-html5/imports/api/slides/server/modifiers/changeCurrentSlide.js (hunk 10 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/api/slides/server/modifiers/changeCurrentSlide.js]: removed

File: bigbluebutton-html5/imports/ui/components/Tabs/TabsView.jsx (hunk 11 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/Tabs/TabsView.jsx]: fixed

File: bigbluebutton-html5/imports/ui/components/Tabs/TabsView.jsx (hunk 12 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/Tabs/TabsView.jsx]: fixed

File: bigbluebutton-html5/imports/ui/components/Tabs/components/Slide/SlideView.jsx (hunk 13 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/Tabs/components/Slide/SlideView.jsx]: fixed

File: bigbluebutton-html5/imports/ui/components/Tabs/components/TabPresentationContent/TabPresentationContent.jsx (hunk 14 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/Tabs/components/TabPresentationContent/TabPresentationContent.jsx]: removed

File: bigbluebutton-html5/imports/ui/components/Tabs/components/TabVideoContent/TabVideoContent.jsx (hunk 15 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/Tabs/components/TabVideoContent/TabVideoContent.jsx]: fixed

File: bigbluebutton-html5/imports/ui/components/app/AppView.jsx (hunk 16 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/app/AppView.jsx]: fixed

File: bigbluebutton-html5/imports/ui/components/common/Button/Button.jsx (hunk 17 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/common/Button/Button.jsx]: fixed

File: bigbluebutton-html5/imports/ui/components/fullscreen-button/FullscreenButtonContainer.jsx (hunk 18 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/fullscreen-button/FullscreenButtonContainer.jsx:15]: fixed

File: bigbluebutton-html5/imports/ui/components/meeting-ended/component.jsx (hunk 19 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/meeting-ended/component.jsx]: removed

File: bigbluebutton-html5/imports/ui/components/presentation/component.jsx (hunk 20 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/presentation/component.jsx:443]: added

File: bigbluebutton-html5/imports/ui/components/presentation/presentation-top-toolbar/PresentationTopToolbarConatiner.jsx (hunk 21 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/presentation/presentation-top-toolbar/PresentationTopToolbarConatiner.jsx]: removed

File: bigbluebutton-html5/imports/ui/components/presentation/presentation-top-toolbar/PresentationTopToolbarConatiner.jsx (hunk 22 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/presentation/presentation-top-toolbar/PresentationTopToolbarConatiner.jsx:56]: added

File: bigbluebutton-html5/imports/ui/components/presentation/service.js (hunk 23 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/presentation/service.js:50]: added

File: bigbluebutton-html5/imports/ui/components/presentation/service.js (hunk 24 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/presentation/service.js:204]: added

File: bigbluebutton-html5/imports/ui/components/presentation/service.js (hunk 25 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/presentation/service.js:212]: added

File: bigbluebutton-html5/imports/ui/components/video-provider/video-button/VideoButtonView.jsx (hunk 26 above)
Comment (Jigar-bhaliya) [reply bigbluebutton-html5/imports/ui/components/video-provider/video-button/VideoButtonView.jsx:49]: added

**Pull request merged**